### PR TITLE
Timestamp support

### LINKED ISSUE #17: Support timestamping
It would be nice to support an option such as `:timestamp: 12520` which appends `?t=12520` to the embed URL.

### PR DESCRIPTION
**Pull request by Robpol86**
Adding optional parameter `:timestamp:` for both YouTube and Vimeo
videos.

Fixes https://github.com/sphinx-contrib/youtube/issues/17

### CODE REVIEW

**Comment by Robpol86** [comment]
Hmm the youtube part doesn't seem to work actually. I was fooled by YouTube since it caches the last time I viewed the video. Tested in incognito mode and it seems to ignore timestamping. Gonna close this for now until I can figure something out.

**Comment by Robpol86** [comment]
Fixed, need to use "start" instead of "t".

**Review by dham — CHANGES_REQUESTED** [review]
Thanks for this nice contribution. I just have two concerns. One is that the timestamp seems to be dropped in LaTeX output, and the other is  what would happen if someone wanted to add more of the URL parameters that YouTube and/or Vimeo support.

I think that both of these can be fixed fairly easily. The first thing to do is to lift the processing of `:timestamp:` into a URL parameter from the html handler up to `Video.run`. This would enable this code to be shared by different backends (because it's not backend-specific). Then, the argument to `_node` could just be called something like `url_parameters`. This indicates that any future parameters should be handled in `run` and just added to this argument.

This approach could then be supported in LaTeX output by simply giving `\sphinxcontribyoutube` a third argument to contain the url parameters.

Cheers,

David

**Comment by akihironitta** [comment]
I'm considering utilising this extension in https://github.com/PyTorchLightning/pytorch-lightning/pull/7688. Any updates on this PR?

**Comment by Robpol86** [comment]
I'm currently working on another project but I plan on getting back to this some time next week.

File: README.rst [hunk 1]
```diff
@@ -31,14 +31,23 @@ parameters "aspect", "width", and "height" may optionally be provided::
...
+
+In LaTeX output, the following code will be emitted for YouTube::
+
+    \sphinxcontribyoutube{https://youtu.be/}{oHg5SJYRHA0}{?start=300}
 
 The user may customise the rendering of the URL by defining this command in 
 the premble. If they do not, then the default definition is used::
 
-    \newcommand{\sphinxcontribvimeo}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2}}\end{center}\end{quote}}
+    \newcommand{\sphinxcontribvimeo}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2#3}}\end{center}\end{quote}}
```
Comment (dham) [inline README.rst]: needs to be `[3]` to indicate that the command takes 3 parameters.

File: sphinxcontrib/youtube/utils.py [hunk 2]
```diff
@@ -88,9 +89,9 @@ def visit_video_node_latex(self, node, platform, platform_url):
     macro = r"\sphinxcontrib%s" % platform
     if macro not in self.elements["preamble"]:
         self.elements["preamble"] += r"""
-        \newcommand{%s}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2}}\end{center}\end{quote}}
+        \newcommand{%s}[2]{\begin{quote}\begin{center}\fbox{\url{#1#2#3}}\end{center}\end{quote}}
```
Comment (dham) [inline sphinxcontrib/youtube/utils.py]: As above, needs to be `[3]`

**Review by dham — CHANGES_REQUESTED** [review]
Thanks. I think this is basically good. I just found one tiny bug with parameter numbers.

**Comment by Robpol86** [comment]
Good catch. Fixed

**Comment by i0bs** [comment]
@dham bumping.

**Pull request merged**